### DISCUSSION TITLE
`dotnet new` to create a solution instead of a project

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/runtime#58753] [Q&A] [answered]
Hi,

I have read https://docs.microsoft.com/en-us/dotnet/core/tutorials/cli-templates-create-project-template and I can install a template and `dotnet new <templateName>` works for me for a single C# project.

How can I make it more general and use `dotnet new <something>` to create a new .NET solution containing (say) 2 projects?

Thank you!

**Comment by huoyaoyuan** [comment, marked as answer]
You can use `dotnet new sln` and `dotnet sln add` command to create such solution.

Solution file is somehow problematic today. It's quite painful to support it.

**Reply by MartyIX** [reply]
Thank you for the answer!

Do I understand it correctly that one needs to do something like this:

```powershell
# See https://stackoverflow.com/questions/42730877/net-core-when-to-use-dotnet-new-sln/42730946#42730946
dotnet new sln --name FooBar
dotnet new console --name Foo --output Foo
dotnet new console --name Bar --output Bar
dotnet sln add .\Foo\Foo.csproj
dotnet sln add .\Bar\Bar.csproj
dotnet restore
dotnet build FooBar.sln
```

to create a single solution with two (C#) projects?

**Reply by huoyaoyuan** [reply]
It looks no problem
